desc/geometry/curve.py

```diff
@@ -770,7 +770,7 @@ def knots(self):
     @knots.setter
     def knots(self, new):
         if len(new) == len(self.knots):
-            knots = jnp.atleast_1d(new)
+            knots = jnp.atleast_1d(jnp.asarray(new))
```

Comment:
jax requires the inputs to the atleast functions to be jnp arrays in newer versions?

Reply:
yeah they've been steadily enforcing jnp arrays to be used for all jnp functions for  a while